test/ClangImporter/print-module-map-paths.swift

```diff
@@ -0,0 +1,38 @@
+// RUN: %empty-directory(%t)
+// RUN: mkdir -p %t/resources/linux/armv7 %t/sdk/usr/include %t/sdk/usr/lib/swift/linux/armv7
+// RUN: touch %t/sdk/usr/include/{inttypes,stdint,unistd}.h
+
+// RUN: touch %t/resources/linux/armv7/{SwiftGlibc.h,glibc.modulemap}
+// RUN: touch %t/sdk/usr/lib/swift/linux/armv7/{SwiftGlibc.h,glibc.modulemap}
+// RUN: %swift %s -typecheck -parse-stdlib -dump-clang-diagnostics -target armv7-unknown-linux-gnueabihf -sdk %t/sdk -resource-dir %t/resources 2>&1 | %FileCheck -check-prefix=CHECK-LINUX %s
+
+// RUN: cp %S/../../stdlib/public/Cxx/{cxxshim/libcxxshim.modulemap,libstdcxx/libstdcxx.h,libstdcxx/libstdcxx.modulemap} %t/resources/linux
+// RUN: if [ %target-os == "linux-gnu" ]; then %swift %s -typecheck -parse-stdlib -dump-clang-diagnostics -resource-dir %t/resources -cxx-interoperability-mode=default 2>&1 | %FileCheck -check-prefix=CHECK-LINUX-CXX %s; fi
+
+// RUN: mkdir -p %t/resources/android/aarch64 %t/sdk/usr/lib/swift/android/aarch64
+// RUN: cp %S/../../stdlib/public/Platform/{SwiftAndroidNDK.h,SwiftBionic.h,android.modulemap} %t/resources/android/aarch64
+// RUN: cp %S/../../stdlib/public/Platform/{SwiftAndroidNDK.h,SwiftBionic.h,android.modulemap} %t/sdk/usr/lib/swift/android/aarch64
+// RUN: %swift %s -typecheck -parse-stdlib -dump-clang-diagnostics -target aarch64-unknown-linux-android -sdk %t/sdk -resource-dir %t/resources 2>&1 | %FileCheck -check-prefix=CHECK-ANDROID %s
+
+// RUN: cp %S/../../stdlib/public/Cxx/cxxshim/libcxxshim.modulemap %t/resources/android
+// RUN: cp %S/../../stdlib/public/Cxx/cxxshim/libcxxshim.modulemap %t/sdk/usr/lib/swift/android
+// RUN: %swift %s -typecheck -parse-stdlib -dump-clang-diagnostics -target aarch64-unknown-linux-android -sdk %t/sdk -resource-dir %t/resources -cxx-interoperability-mode=default 2>&1 | %FileCheck -check-prefix=CHECK-ANDROID-CXX %s
+
+// CHECK-LINUX: clang importer redirected file mappings:
+// CHECK-LINUX-NEXT: mapping real file '{{.*}}/resources/linux/armv7/glibc.modulemap' to virtual file '{{.*}}/sdk/usr/include/module.modulemap'
+// CHECK-LINUX-NEXT: mapping real file '{{.*}}/resources/linux/armv7/SwiftGlibc.h' to virtual file '{{.*}}/sdk/usr/include/SwiftGlibc.h'
+
+// CHECK-LINUX-CXX: mapping real file '{{.*}}/resources/linux/libstdcxx.h' to virtual file '{{.*}}/usr/include/c++/{{.*}}/libstdcxx.h'
+// CHECK-LINUX-CXX: clang importer overriding file '{{.*}}/usr/include/c++/{{.*}}/module.modulemap' with the following contents:
+// CHECK-LINUX-CXX-NEXT: --- libstdcxx.modulemap ---
+// CHECK-LINUX-CXX: Currently libstdc++ does not have a module map. To work around
+// CHECK-LINUX-CXX-NEXT: this, Swift provides its own module map for libstdc++.
+// CHECK-LINUX-CXX: header "libstdcxx.h"
+// CHECK-LINUX-CXX: clang importer driver args: {{.*}}'-fmodule-map-file={{.*}}resources/linux/libcxxshim.modulemap'
```

Comment:
I was surprised this was used with libstdc++, but this `libcxxshim.modulemap` isn't in the static linux Musl SDK that ships libc++.

@al45tair, is this needed for all C++ stdlibs, and if so, why doesn't your Musl SDK ship it?

Reply:
libc++ has its own module maps already; it shouldn't need shimming.

Reply:
Ah sorry I missed this comment. These shims are used with all C++ stdlibs that we support, the lack of them can cause compiler crashes.

Reply:
Ah, interesting. In which case we probably need libcxxshim.modulemap after all.